### DISCUSSION TITLE
How to prompt sign-in from the user?

### DISCUSSION THREAD

**Discussion by mj-schmitz** [hyochan/react-native-iap#2234] [Q&A]
I'm working on creating a restore button within the app.  I have the restore button get the purchases of the user, using getAvailablePurchases(), however this of course is an empty array when the user is not signed in when they get to the subscription screen.

How can I trigger the user to sign in in this case?  It seems the only function when using useIAP context is the requestSubscription method.  Any ideas on how to trigger the user to sign in within the following function:

    const handleRestore = () => {
        getAvailablePurchases().then(() => {
            console.log(`Available Purchases:  ${JSON.stringify(availablePurchases, null, 1)} ` );
            availablePurchases.map((purchase)=> {
                if(purchase.productId === 'id'){
                    setRestored(true);
                    console.log('handleRestore - Restoring Purchase');
                }
            });
        }).catch( (err) => {
            console.log(err,'err');
        });
    };

**Comment by andresesfm** [comment]
take a look at `getReceiptIOS`.
